test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs

```diff
@@ -1478,6 +1478,28 @@ public void CollectFrameworkReferences_WithTransitiveFrameworkReferences_Exclude
             }
         }
 
+        [PlatformTheory(Platform.Linux)]
+        [InlineData("worker")]
+        [InlineData("mstest")]
+        public void Dotnet_New_Template_Restore_Success(string template)
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var projectName = new DirectoryInfo(pathContext.SolutionRoot).Name;
+                var solutionDirectory = pathContext.SolutionRoot;
+
+                // Act
+                CommandRunnerResult newResult = _msbuildFixture.RunDotnet(solutionDirectory, "new " + template);
```

Comment:
`_msbuildFixture.CreateDotnetNewProject`

Reply:
This one doesn't do `restore` after creating new template for `worker`.